### PR TITLE
DT-1345 Added Cancellation of confirmed booking.

### PR DESCRIPTION
**Pull request by preetamnpr**

### CODE REVIEW

File: booking/src/main/java/org/dcsa/conformance/standards/booking/action/StateChangingBookingAction.java [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+import org.dcsa.conformance.core.check.JsonSchemaValidator;
+import org.dcsa.conformance.core.check.ResponseStatusCheck;
+import org.dcsa.conformance.core.check.UrlPathCheck;
 import org.dcsa.conformance.core.traffic.ConformanceExchange;
+import org.dcsa.conformance.core.traffic.HttpMessageType;
+import org.dcsa.conformance.standards.booking.checks.BookingChecks;
 import org.dcsa.conformance.standards.booking.checks.ScenarioType;
+import org.dcsa.conformance.standards.booking.party.BookingRole;
+
+import javax.swing.*;
```
Comment (jkosternl) [inline booking/src/main/java/org/dcsa/conformance/standards/booking/action/StateChangingBookingAction.java:17]: I guess this import is wrong. We're not using Java Swing elements..

Comment (preetamnpr) [reply booking/src/main/java/org/dcsa/conformance/standards/booking/action/StateChangingBookingAction.java:17]: I dont know when this got imported.  will remove the unused imports

File: booking/src/main/java/org/dcsa/conformance/standards/booking/party/BookingShipper.java [hunk 2]
```diff
@@ -146,11 +147,25 @@ private void sendCancelBookingRequest(JsonNode actionPrompt) {
...
     addOperatorLogEntry("Sent a cancel booking request of '%s'".formatted(cbrr));
   }
 
+  private void sendConfirmedBookingCancellationRequest(JsonNode actionPrompt) {
+    log.info("Shipper.sendConfirmedBookingCancellationRequest(%s)".formatted(actionPrompt.toPrettyString()));
+    String cbr = actionPrompt.path("cbr").asText();
+    syncCounterpartPatch(
+      "/v2/bookings/%s".formatted(cbr),
+      Collections.emptyMap(),
+      new ObjectMapper()
```
Comment (nt-gt) [inline booking/src/main/java/org/dcsa/conformance/standards/booking/party/BookingShipper.java:161]: Use `OBJECT_MAPPER`

**Review by nt-gt — APPROVED** [review]

**Comment by douradofrazer** [comment]
@CodiumAI-Agent /review
--pr_reviewer.inline_code_comments=true
--pr_reviewer.require_score_review=true

**Pull request merged**